teacher.rb

```diff
@@ -0,0 +1,14 @@
+require_relative('./person')
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, age, name, parent_permission)
+    @specialization = specialization
+    super(age, name, parent_permission)
+  end
```

Comment:
- [ ] Since name and parent_permission are optional and have default values of "Unknown" and true you have to write it while initializing:

```ruby
 def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

You should also put super right after initialization as shown in example.

Reply:
Thank you @ShoiraTa for reviewing. I have fixed this issue.